frontend/src/components/teacher/dashboard/ClassroomsSection.tsx

```diff
@@ -26,21 +25,9 @@ const ClassroomsSection = (): ReactElement => {
   const { loading, error, data } = useClassDataQuery(QUERY_DATA_OPTIONS);
 
   return (
-    <Box flex="1" w="100%">
-      {loading && (
-        <Center flex="1" margin="15%" mt="calc(15% - 2rem)">
-          <LoadingState />
-        </Center>
-      )}
-      {error && (
-        <Box height="100%" mt={2}>
-          <ErrorState />
-        </Box>
-      )}
-      {data &&
-        !error &&
-        !loading &&
-        (data.length === 0 ? (
+    <QueryStateHandler error={error} loading={loading}>
+      <Box mt={8}>
+        {data?.length === 0 ? (
```

Comment:
nit: maybe invert the order and use the logic that we had in other places: `data?.length ? <RealComponent /> : <EmptyState />`